### DISCUSSION TITLE
What is multiple copies of React Aria?

### DISCUSSION THREAD

**Discussion by gavinliu6** [adobe/react-spectrum#3430] [Q&A] [answered]
There is such a description in the official documentation:

> When using SSR, only a single copy of React Aria can be on the page at a time. This is in contrast to client-side rendering, where multiple copies from different parts of an app can coexist. Internally, many components rely on auto-generated ids to link related elements via ARIA attributes. These ids typically use a randomly generated seed plus an incrementing counter to ensure uniqueness even when multiple instances of React Aria are on the page. With SSR, we need to ensure that these ids are consistent between the server and client. This means the counter resets on every request, and we use a consistent seed. Due to this, multiple copies of React Aria cannot be supported because the auto-generated ids would conflict.

Can anyone help explain what _copy of React Aria_ means here? It's better to give some examples.

**Comment by snowystinger** [comment, marked as answer]
It's referring to either multiple versions or copies of any of the react aria modules. Much like you can't have two copies of React on a page.

**Reply by gavinliu6** [reply]
I get it, thanks.
